sys/shell/shell.c

```diff
@@ -343,7 +332,7 @@ __attribute__((weak)) void shell_post_command_hook(int ret, int argc,
 
 static inline void print_prompt(void)
 {
-    if (PROMPT_ON) {
+    if (!IS_ACTIVE(CONFIG_SHELL_NO_PROMPT) && !IS_ACTIVE(SHELL_NO_PROMPT)) {
         putchar('>');
```

Comment:
This would be a functional change, so I would completely understand if you say no, but provided that `>` sometimes makes problems when using `riotctrl`'s `ShellInteraction` when there is a `>` in the output, would it make sense to define`CONFIG_SHELL_PROMPT_CHAR` instead of `CONFIG_SHELL_NO_PROMPT` and define `SHELL_NO_PROMPT == (IS_ACTIVE(CONFIG_SHELL_PROMPT_CHAR)`?

Reply:
Added an option to modify the char prompt. Can you check if it's ok?

Reply:
It seems that this feature has some issues with escaping special character (e.g. `$`). I will rollback 465a0c6 for now, if we can come up with a better solution a follow up PR can be opened.